lib/python/src/bailo/helper/release.py

```diff
@@ -171,12 +230,26 @@ def upload(self, path: str, data: BytesIO | None = None) -> str:
                 name = path
 
             with open(path, "rb") as f:
-                res = self.client.simple_upload(self.model_id, name, f).json()
+                data = BytesIO(f.read())
 
             if is_zip:
                 os.remove(path)
+
+        old_file_position = data.tell()
+        data.seek(0, os.SEEK_END)
+        size = data.tell()
+        data.seek(old_file_position, os.SEEK_SET)
+
+        if no_color == 1:
+            colour = "white"
         else:
-            res = self.client.simple_upload(self.model_id, name, data).json()
+            colour = "blue"
+
+        with tqdm(
+            total=size, unit="B", unit_scale=True, unit_divisor=1024, postfix=f"uploading {name}", colour=colour
+        ) as t:
+            wrapped_buffer = CallbackIOWrapper(t.update, data, "read")
+            res = self.client.simple_upload(self.model_id, name, wrapped_buffer).json()
```

Comment:
The progress bar seems to complete even if the request fails. I tested this with uploading two of the same file name.

Reply:
Done